src/data/calculate.ts

```diff
@@ -290,29 +291,43 @@ export const calculatePersonRiskEach = (
       })
 
     if (
-      // TODO: Support vaccinated risk for other Risk Profiles.
-      data.riskProfile !== 'average' ||
+      !VaccineModifiableRiskProfiles.includes(data.riskProfile) ||
       data.unvaccinatedPrevalenceRatio === null ||
       data.averageFullyVaccinatedMultiplier === null
     ) {
       return unadjustedRisk
     }
 
-    switch (data.theirVaccine) {
-      case 'vaccinated':
-        return (
-          unadjustedRisk *
-          data.unvaccinatedPrevalenceRatio *
-          data.averageFullyVaccinatedMultiplier
-        )
-      case 'unvaccinated':
-        return unadjustedRisk * data.unvaccinatedPrevalenceRatio
-      // falls through
-      case 'undefined':
-        return unadjustedRisk
-      default:
-        console.error(`Unrecognized vaccination state: ${data.theirVaccine}`)
-        return null
+    if (data.riskProfile === 'average') {
+      switch (data.theirVaccine) {
+        case 'vaccinated':
+          return (
+            unadjustedRisk *
+            data.unvaccinatedPrevalenceRatio *
+            data.averageFullyVaccinatedMultiplier
+          )
+        case 'unvaccinated':
+          return unadjustedRisk * data.unvaccinatedPrevalenceRatio
+        case 'undefined':
+          return unadjustedRisk
+        default:
+          console.error(`Unrecognized vaccination state: ${data.theirVaccine}`)
+          return null
+      }
+    } else {
+      // These are risk profiles that were set up for unvaccinated people.
+      switch (data.theirVaccine) {
+        case 'vaccinated':
+          return unadjustedRisk * data.averageFullyVaccinatedMultiplier
+        case 'unvaccinated':
+          return unadjustedRisk
+        // falls through
+        case 'undefined':
+          return unadjustedRisk / data.unvaccinatedPrevalenceRatio
```

Comment:
I think this should be `* percentVaccinated * averageFullyVaccinatedMultiplier`?

Reply:
I basically just followed the existing relationships. If risk profile is average (unadjusted risk means "don't know if they're vaccinated"), then [this line](https://github.com/microcovid/microcovid/pull/932/files/05efc0056efde3345c51d2530aca71ac00225cd3#diff-7d933aaf4cb288c3d417f891c2089c9cd5dbb58ddeb497b5f177b01ae7ea3809L308) says that `unvaccinated = [risk for unknown vax status] * data.unvaccinatedPrevalenceRatio`.

I think the original logic stands and it is correct as written; see [this comment](https://github.com/microcovid/microcovid/pull/932/files/05efc0056efde3345c51d2530aca71ac00225cd3#diff-093196c543f2689ed8ed57c57bb46ce008529f882556657678049ccbc38497f9R360).

Reply:
Oh, `* (averageVaccine * percentVaccinated + 1 * (1 - percentVaccinated))` drops partially vaxxed, so the formulation you wrote is better.

Reply:
Could you leave a comment here?

// data.unvaccinatedPrevalenceRatio = population / risk_sum -> 1 / data.unvaccinatedPrevalenceRatio  = average vaccine modifier of the population

Reply:
I briefly explained what it was and pointed source code readers to the original calculation.